csrc/scheduler/vectorize_helper.cpp

```diff
@@ -400,44 +375,50 @@ std::vector<IterDomain*> ContiguousInnerDimensionsMapper::projectId(
     if (it == frontier.end()) {
       return;
     }
-
     auto pos = std::distance(frontier.begin(), it);
-    if (resize_in_pad_.count(resize_op) != 0) {
-      // resize created by PadOp.
 
-      // project resize op to frontier.
-      frontier[pos] = id_to;
-      // clear left of resize, since those are no long contiguous.
-      frontier.erase(frontier.begin(), it);
+    // project resize op to frontier.
+    frontier[pos] = id_to;
+    // clear left of resize, since those are no long contiguous.
+    frontier.erase(frontier.begin(), it);
 
-      if (recording_) {
-        // TODO: support negative resize extent.
-        //
-        // Limit current support to only positive resize extent for now. So we
-        // only consider the pad_extent, which becomes the real buffer on
-        // output. Hence we do GCD among padded extent as well as extent of the
-        // id_from. Note since we are taking the GCD here, I don't think using
-        // id_from or id_to makes a difference.
-        auto consumer_factor = getProjectedExtent(id_from);
-        auto comp = [](Val* factor, Val* extent) {
-          return SimplifyingIrBuilder::whereExpr(
-              SimplifyingIrBuilder::eqExpr(
-                  extent, extent->container()->zeroVal()),
-              factor,
-              // for extent < 0, we'll take max(1, extent). Because of the gcd,
-              // This is effectively excluding the resize id from vectorization.
-              SimplifyingIrBuilder::gcdExpr(
-                  factor,
-                  SimplifyingIrBuilder::maxExpr(
-                      extent->container()->oneVal(), extent)));
-        };
-        consumer_factor = comp(consumer_factor, resize_op->leftExpand());
-        consumer_factor = comp(consumer_factor, resize_op->rightExpand());
-        addProjectedExtent(id_to, consumer_factor);
+    if (recording_) {
+      // we need to check slice offset at this point.
+      auto projected_extent = getProjectedExtent(id_from);
+
+      std::vector<Val*> expands = {
+          resize_op->leftExpand(), resize_op->rightExpand()};
+      if (!p2c) {
+        // reverse pad extent for c2p propagation
+        std::for_each(expands.begin(), expands.end(), [](Val*& val) {
+          val = SimplifyingIrBuilder::negExpr(val);
+        });
```

Comment:
Unused?

Reply:
Note the lambda signature, `Val*&`
This one is modifying in place...

I'll update the comment to note that.

Reply:
Still unclear to me. I don't see any usage of `expands` after this part. What am I missing?

Reply:
no you didn't, I missed to apply the expands onto projected_extent.

NOTE for myself, adding a unit test to verify this.

Reply:
So, this wasn't necessary after all?

Reply:
I just figured out an easier way to do this. :face_with_head_bandage: 

I'm just trying to get the destination size and somehow I was obsessed with computing it from the source. Forgot that I can just do `id_to->extent()` instead.